src/algebra/smul_regular.lean

```diff
@@ -38,6 +38,9 @@ section has_scalar
 
 variables [has_scalar R M] [has_scalar R S] [has_scalar S M] [is_scalar_tower R S M]
 
+lemma eq_of_smul_eq {x y : M} (ra : is_smul_regular M a) (h : a • x = a • y) : x = y :=
+ra h
```

Comment:
I'm not convinced this lemma is worth having vs just using `ra h` or `ra.eq_iff` directly - if there is a benefit to this form over that, can you add it in a comment?

Reply:
It's just for ease of discovery for people who don't know what `injective` is defeq to. In fact it turned out that I did not use this file (I discovered later `smul_left_injective` which is in a different file and is enough for my needs), but still I left the modifications I have made to this file because I still thought they could be helpful. I can remove this lemma if you like, no problem, tell me!

Reply:
In fact I have removed the modifications to this file, as they are not relevant for the PR, for easier review.